dynosaur/tests/pass/handle-generics-on-trait-def.stdout

```diff
@@ -34,9 +34,15 @@ impl<T, DYNOSAUR: MyTrait<T>> ErasedMyTrait<T> for DYNOSAUR {
         Box::pin(<Self as MyTrait<T>>::foo(self, x))
     }
 }
-struct DynMyTrait<'dynosaur, T, Item> {
-    ptr: *mut (dyn ErasedMyTrait<T, Item = Item> + 'dynosaur),
+struct DynMyTrait<'dynosaur_struct, T, Item> {
+    ptr: *mut (dyn ErasedMyTrait<T, Item = Item> + 'dynosaur_struct),
     owned: bool,
 }
+impl<'dynosaur_struct, T, Item> MyTrait<T> for
+    DynMyTrait<'dynosaur_struct, T, Item> {
+    const N: i32 = <Self as MyTrait<T>>::N;
```

Comment:
This const is referring to itself?

We don't expect this to compile anyway, but it seems like this would make for some confusing output.

Reply:
As you have said it won't compile anyway, I was thinking what to do with it given that it won't compile. With the current code the output doesn't look bad ...

```
error[E0038]: the trait `ErasedMyTrait` cannot be made into an object
  --> src/main.rs:37:16
   |
37 |     ptr: *mut (dyn ErasedMyTrait<T, Item = Item> + 'dynosaur_struct),
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `ErasedMyTrait` cannot be made into an object
   |
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
  --> src/main.rs:14:11
   |
13 | trait ErasedMyTrait<T> {
   |       ------------- this trait cannot be made into an object...
14 |     const N: i32;
   |           ^ ...because it contains this associated `const`
   = help: consider moving `N` to another trait

error[E0038]: the trait `ErasedMyTrait` cannot be made into an object
  --> src/main.rs:44:41
   |
44 |     async fn foo(&self, x: &T) -> i32 { unsafe { &*self.ptr }.foo(x).await }
   |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `ErasedMyTrait` cannot be made into an object
   |
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
  --> src/main.rs:14:11
   |
13 | trait ErasedMyTrait<T> {
   |       ------------- this trait cannot be made into an object...
14 |     const N: i32;
   |           ^ ...because it contains this associated `const`
   = help: consider moving `N` to another trait
```

If I remove the const entirely I think is more confusing ...

```
error[E0038]: the trait `ErasedMyTrait` cannot be made into an object
  --> src/main.rs:37:16
   |
37 |     ptr: *mut (dyn ErasedMyTrait<T, Item = Item> + 'dynosaur_struct),
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `ErasedMyTrait` cannot be made into an object
   |
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
  --> src/main.rs:14:11
   |
13 | trait ErasedMyTrait<T> {
   |       ------------- this trait cannot be made into an object...
14 |     const N: i32;
   |           ^ ...because it contains this associated `const`
   = help: consider moving `N` to another trait

error[E0038]: the trait `ErasedMyTrait` cannot be made into an object
  --> src/main.rs:43:41
   |
43 |     async fn foo(&self, x: &T) -> i32 { unsafe { &*self.ptr }.foo(x).await }
   |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `ErasedMyTrait` cannot be made into an object
   |
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
  --> src/main.rs:14:11
   |
13 | trait ErasedMyTrait<T> {
   |       ------------- this trait cannot be made into an object...
14 |     const N: i32;
   |           ^ ...because it contains this associated `const`
   = help: consider moving `N` to another trait

error[E0046]: not all trait items implemented, missing: `N`
  --> src/main.rs:40:1
   |
8  |       const N: i32;
   |       ------------ `N` from trait
...
40 | / impl<'dynosaur_struct, T, Item> MyTrait<T> for
41 | |     DynMyTrait<'dynosaur_struct, T, Item> {
   | |_________________________________________^ missing `N` in implementation
```